ocaml/lang/correctness/useless_eq.yaml

```diff
@@ -0,0 +1,6 @@
+rules:
+- id: useless-equal
+  pattern: $X = $X
+  message: This is always true.
+  languages: [ocaml]
+  severity: ERROR
```

Comment:
Is this something people do? Normally it's done to check that a float is not a NaN. OCaml has now a `Float.is_nan` to take care of it, so there's no longer a justification for tests like `x <> x`. If compatibility with old versions of ocaml is important, we can suggest they write their own `is_nan` function e.g.
```ocaml
let is_nan x =
  classify_float x = FP_nan
```

Reply:
good idea. Actually we have a similar rule for python that also suggest a special function to check for NaN.

And yes people do write equal expressions ... usually because they made a  typo. This rule found a surprisingly big number of bugs in open source projects ... you would be surprised.

Reply:
@mjambon This is a common oops in webapps:

```python
user_id == user_id
```

when someone meant

```python
user.id == user_id
```